ResearchKit/Common/ORKWaitStepView.m

```diff
@@ -0,0 +1,153 @@
+        [constraints addObjectsFromArray:
+         [NSLayoutConstraint constraintsWithVisualFormat:@"V:[_textLabel]-[_progressView]|"
+                                                 options:NSLayoutFormatAlignAllCenterX
+                                                 metrics:nil
+                                                   views:NSDictionaryOfVariableBindings(_progressView, _textLabel)]];
+        
+    } else if (_activityIndicatorView) {
+        
+        [constraints addObject:
+         [NSLayoutConstraint constraintWithItem:_activityIndicatorView
+                                      attribute:NSLayoutAttributeCenterX
+                                      relatedBy:NSLayoutRelationEqual
+                                         toItem:self
+                                      attribute:NSLayoutAttributeCenterX
+                                     multiplier:1.0
+                                       constant:0.0]];
+        [constraints addObjectsFromArray:
+         [NSLayoutConstraint constraintsWithVisualFormat:@"H:|-(>=0)-[_activityIndicatorView]-(>=0)-|"
+                                                 options:NSLayoutFormatAlignAllBaseline
+                                                 metrics:nil
+                                                   views:NSDictionaryOfVariableBindings(_activityIndicatorView)]];
+        [constraints addObjectsFromArray:
+         [NSLayoutConstraint constraintsWithVisualFormat:@"V:[_textLabel]-[_activityIndicatorView]|"
+                                                 options:NSLayoutFormatAlignAllCenterX
+                                                 metrics:nil
+                                                   views:NSDictionaryOfVariableBindings(_activityIndicatorView, _textLabel)]];
+    }
+    
+    [NSLayoutConstraint activateConstraints:constraints];
+}
+
+- (void)setIndicatorMask:(ORKProgressIndicatorMask)indicatorMask {
+    if (_indicatorMask != indicatorMask) {
+        _indicatorMask = indicatorMask;
+        
+        switch (_indicatorMask) {
+            case ORKProgressIndicatorMaskProgressBar:
+                [_activityIndicatorView removeFromSuperview];
+                _activityIndicatorView = nil;
+                _progressView = [UIProgressView new];
+                _progressView.translatesAutoresizingMaskIntoConstraints = NO;
+                [self addSubview:_progressView];
+                break;
+            case ORKProgressIndicatorMaskIndeterminate:
+                [_progressView removeFromSuperview];
+                _progressView = nil;
+                _activityIndicatorView = [[ORKProgressView alloc] init];
+                _activityIndicatorView.translatesAutoresizingMaskIntoConstraints = NO;
+                [self addSubview:_activityIndicatorView];
+                break;
+            default:
+                break;
+        }
+        
+        [self setUpConstraints];
```

Comment:
`[self setUpConstraints]` should only be called once from `init`, `setUpConstraints` multiple times could result duplicated constraints.